packages/fuels/src/cli/commands/dev/index.ts

```diff
@@ -0,0 +1,64 @@
+import * as chokidar from 'chokidar';
+import { globSync } from 'glob';
+
+import type { FuelsConfig } from '../../types';
+import { error, log } from '../../utils/logger';
+import { build } from '../build';
+import { deploy } from '../deploy';
+
+import { startFuelCore } from './startFuelCore';
+
+export const fileHandlers: {
+  watcher?: chokidar.FSWatcher;
+} = {};
+
+export const buildAndDeploy = async (config: FuelsConfig) => {
+  await build(config);
+  return deploy(config);
+};
+
+export const changeListener = (config: FuelsConfig) => async (path: string) => {
+  log(`\nFile changed: ${path}`);
+  await buildAndDeploy(config);
+};
+
+export async function dev(config: FuelsConfig) {
+  // here we inject a `providerUrl` into the config if necessary
+  if (config.autoStartFuelCore) {
+    const client = await startFuelCore(config);
+    // eslint-disable-next-line no-param-reassign
+    config.providerUrl = client.providerUrl;
+  }
+
+  const { contracts, scripts, predicates, chainConfig, workspace } = config;
+
+  const projectDirs = [contracts, scripts, predicates, chainConfig].flat();
+
+  if (workspace) {
+    projectDirs.push(workspace);
```

Comment:
Is it necessary to add the `workspace` path here? It's already accounted for in the `contracts`/`scripts`/`predicates` that are built inside of the `loadConfig` function.

Reply:
Yes, it needs to be considered conditionally.

The fact it is not has masked the problem with all [other](https://github.com/FuelLabs/fuels-ts/pull/1291#discussion_r1367718190) paths — _being transformed into broken globals_.

Reply:
https://github.com/FuelLabs/fuels-ts/blob/bdb47a5c70c3d397256330527691faecdf6fcfe0/packages/fuels/src/cli/commands/dev/index.ts#L70-L76